src/plugins/discover/public/application/main/utils/get_switch_index_pattern_app_state.ts

```diff
@@ -28,19 +28,20 @@ export function getSwitchIndexPatternAppState(
       )
     : currentColumns;
   const columns = nextColumns.length ? nextColumns : [];
-  // when switching from an index pattern with timeField to an index pattern without timeField
-  // filter out sorting by timeField in case it is set. index patterns without timeField don't
-  // prepend this field in the table, so in legacy grid you would need to add this column to
-  // remove sorting
-  const nextSort = getSortArray(currentSort, nextIndexPattern).filter((value) => {
-    return nextIndexPattern.timeFieldName || value[0] !== currentIndexPattern.timeFieldName;
-  });
+
+  // remove prev time field sort
+  let nextSort = getSortArray(currentSort, nextIndexPattern).filter(
+    (value) => value[0] !== currentIndexPattern.timeFieldName
+  );
+
+  // append default sorting if indexPattern has time field
+  if (nextIndexPattern.timeFieldName) {
+    nextSort = [[nextIndexPattern.timeFieldName, sortDirection], ...nextSort];
+  }
```

Comment:
This would work well but there's a multisort case to consider. So the user could e.g. sort by `_score` first and `_currentIndexPattern.timeFieldName` second. 

Currently we would remove the second param, and prepend `nextIndexPattern.timeFieldName`. This could lead to confusion. 

So the switching/removing of the sorting by `timeFieldName` should be only done when it's the first field to sort by in `getSortArray(currentSort, nextIndexPattern)`. Also, in this case the direction of the sort should be kept.